src/plugins/intel_npu/src/plugin/npuw/base_sync_infer_request.cpp

```diff
@@ -593,6 +618,59 @@ void ov::npuw::IBaseInferRequest::handle_quant_host_gather(std::size_t idx, RqPt
     }
 }
 
+void ov::npuw::IBaseInferRequest::bind_attention_inputs(std::size_t idx, RqPtr request) {
+    auto& comp_model_desc = m_npuw_model->m_compiled_submodels[real(idx)];
+    if (!comp_model_desc.dynamic) {
+        return;
+    }
+
+    LOG_DEBUG("Binding Attention inputs...");
+    LOG_BLOCK();
+
+    const auto& dynamic = comp_model_desc.dynamic.value();
+    auto& r = request;
+
+    const auto pos_id = m_dynamic_selector->length();
+    if (pos_id == -1) {
+        // Dynamic range couldn't be identified - fallback to the default
+        // (worst case) behavior
+        for (auto&& param : dynamic.params) {
+            const auto& iport = comp_model_desc.compiled_model->inputs()[param.idx];
+            const auto& input = m_dynamic_io[idx].inputs.at(param.idx);
+            r->set_tensor(iport, input);
+        }
+    } else {
+        const auto past_len = m_dynamic_selector->past_length();
+        // Set the past k/v values first
+        for (auto&& param : dynamic.params) {
+            const auto& iport = comp_model_desc.compiled_model->inputs()[param.idx];
+            const auto& input = m_dynamic_io[idx].inputs.at(param.idx);
+            const auto& view = ov::npuw::util::view(input, param.dim, 0, past_len);
+            const auto shape = view->get_shape();
+            const auto do_copy = needs_copy(idx);
+
+            LOG_DEBUG(iport);
+            LOG_BLOCK();
+            if (do_copy && ov::shape_size(shape) > 0) {
```

Comment:
probably introduce a knob to force avoid copy - to highlight some issues without recompilation

Reply:
NPUW_ATTN_NO_COPY is the knob